pyanaconda/modules/subscription/runtime.py

```diff
@@ -360,3 +368,200 @@ def run(self):
+    def _pretty_date(self, date_from_json):
+        """Return pretty human readable date based on date from the input JSON."""
+        # fallback in case of the parsing fails
+        date_string = date_from_json
+        try:
+            # The start/end date in GetPools() output seems to be formatted as
+            # "Locale’s appropriate date representation.".
+            # See bug 1793501 for possible issues with RHSM provided date parsing.
+            date = datetime.datetime.strptime(date_from_json, "%m/%d/%y")
+            # get a nice human readable date
+            date_string = date.strftime("%b %d, %Y")
+        except ValueError:
+            log.warning("subscription: date parsing failed: %s", date_from_json)
+        return date_string
+
+    @classmethod
+    def _parse_subscription_json(cls, subscription_json):
+        """Parse the JSON into list of AttachedSubscription instances.
+
+        The expected JSON is at top level a list of rather complex dictionaries,
+        with each dictionary describing a single subscription that has been attached
+        to the system.
+
+        :param str subscription_json: JSON describing what subscriptions have been attached
+        :return: list of attached subscriptions
+        :rtype: list of AttachedSubscription instances
+        """
+        attached_subscriptions = []
+        try:
+            subscriptions = json.loads(subscription_json)
+        except json.decoder.JSONDecodeError:
+            log.warning("subscription: failed to parse GetPools() JSON output")
+            # empty attached subscription list is better than an installation
+            # ending crash
+            return []
+        # find the list of subscriptions
+        consumed_subscriptions = subscriptions.get("consumed", [])
+        log.debug("subscription: parsing %d attached subscriptions",
+                  len(consumed_subscriptions))
+        # split the list of subscriptions into separate subscription dictionaries
+        for subscription_info in consumed_subscriptions:
+            attached_subscription = AttachedSubscription()
+            # user visible product name
+            attached_subscription.name = subscription_info.get(
+                "subscription_name",
+                _("product name unknown")
+            )
+
+            # subscription support level
+            # - this does *not* seem to directly correlate to system purpose SLA attribute
+            attached_subscription.service_level = subscription_info.get(
+                "service_level",
+                _("unknown")
+            )
+
+            # SKU
+            # - looks like productId == SKU in this JSON output
+            attached_subscription.sku = subscription_info.get(
+                "sku",
+                _("unknown")
+            )
+
+            # contract number
+            attached_subscription.contract = subscription_info.get(
+                "contract",
+                _("Not Available")
```

Comment:
This string is title cased, but the other strings are lower cased. Maybe it could be unified?

Reply:
Good point, let's do that. :)